pysb/simulator/bng.py

```diff
@@ -0,0 +1,227 @@
+        """
+        Simulate a model with BNG's SSA simulator and return the trajectories.
+
+        Parameters
+        ----------
+        tspan: vector-like
+            time span of simulation
+        initials: vector-like, optional
+            initial conditions of model
+        param_values : vector-like or dictionary, optional
+            Values to use for every parameter in the model. Ordering is
+            determined by the order of model.parameters.
+            If not specified, parameter values will be taken directly from
+            model.parameters.
+        n_runs: int
+            number of simulations to run
+        method : str
+            Type of simulation to run. Must be one of:
+
+             * 'ssa' - Stochastic Simulation Algorithm (direct method with
+             propensity sorting)
+             * 'nf' - Stochastic network free simulation with NFsim
+             * 'pla' - Partioned-leaping algorithm (variant of tau-leaping
+             algorithm)
+             * 'ode' - ODE simulation (Sundials CVODE algorithm)
+             * 'hpp' - Hybrid Particle/Population simulation 
+             (https://doi.org/10.1371/journal.pcbi.1003544)
+
+        output_dir : string, optional
+            Location for temporary files generated by BNG. If None (the
+            default), uses a temporary directory provided by the system. A
+            temporary directory with a random name is created within the
+            supplied location.
+        output_file_basename : string, optional
+            This argument is used as a prefix for the temporary BNG
+            output directory, rather than the individual files.
+        cleanup : bool, optional
+            If True (default), delete the temporary files after the
+            simulation is
+            finished. If False, leave them in place. Useful for debugging.
+        additional_args: kwargs, optional
+            Additional arguments to pass to BioNetGen
+
+        """
+        super(BngSimulator, self).run(tspan=tspan,
+                                      initials=initials,
+                                      param_values=param_values
+                                      )
+
+        if method not in self._SIMULATOR_TYPES:
+            raise ValueError("Method must be one of " +
+                             str(self._SIMULATOR_TYPES))
+
+        if method == 'hpp':
+            if population_maps is None:
+                raise ValueError('population_maps need to be supplied')
+            if not isinstance(population_maps, collections.Iterable) or not \
+                    all([isinstance(pm, PopulationMap) for pm in
+                         population_maps]):
+                raise ValueError('population_maps should be a list of '
+                                 'PopulationMap objects')
+
+        # Calculate time intervals and validate
+        if not np.allclose(self.tspan, np.linspace(0, self.tspan[-1],
```

Comment:
It's only NFsim that can only take `t_end` and `n_steps`, not a list of `sample_times`. Will switch to `sample_times` where possible, and limit the check and Exception to NFsim-based simulations. StochKit reference was a mistake from copy/paste; will amend.